LWM2M Generator throws error in console

Steps:
Infomodel > Generate code > choose LWM2M generator

Actual Result:
_Error in console:_
[ERROR] java.lang.IllegalArgumentException: Resource ID <0> conflict for <yValue> - abort Generator!
            at org.eclipse.vorto.codegen.examples.lwm2m.templates.FunctionBlockXmlTemplate.checkResourceIdConflictAndFillSet(FunctionBlockXmlTemplate.java:243)
            at org.eclipse.vorto.codegen.examples.lwm2m.templates.FunctionBlockXmlTemplate.handleProperties(FunctionBlockXmlTemplate.java:303)
            at org.eclipse.vorto.codegen.examples.lwm2m.templates.FunctionBlockXmlTemplate.getContent(FunctionBlockXmlTemplate.java:103)
            at org.eclipse.vorto.codegen.examples.lwm2m.templates.FunctionBlockXmlTemplate.getContent(FunctionBlockXmlTemplate.java:1)
            at org.eclipse.vorto.codegen.api.AbstractTemplateGeneratorTask.generate(AbstractTemplateGeneratorTask.java:24)
            at org.eclipse.vorto.codegen.examples.lwm2m.LWM2MGenerator.generate(LWM2MGenerator.java:35)
            at org.eclipse.vorto.codegen.ui.handler.CodeGeneratorInvocationHandler.evaluate(CodeGeneratorInvocationHandler.java:85)
            at org.eclipse.vorto.codegen.ui.handler.CodeGeneratorInvocationHandler.execute(CodeGeneratorInvocationHandler.java:60)
            at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:295)
            at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:90)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
            at java.lang.reflect.Method.invoke(Unknown Source)
            at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:56)
            at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:252)
            at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:234)
            at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
            at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
            at org.eclipse.core.commands.Command.executeWithChecks(Command.java:493)
            at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:486)
            at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
            at org.eclipse.ui.internal.handlers.LegacyHandlerService.executeCommand(LegacyHandlerService.java:343)
            at org.eclipse.ui.menus.CommandContributionItem.handleWidgetSelection(CommandContributionItem.java:830)
            at org.eclipse.ui.menus.CommandContributionItem.access$21(CommandContributionItem.java:816)
            at org.eclipse.ui.menus.CommandContributionItem$5.handleEvent(CommandContributionItem.java:806)
            at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
            at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4362)
            at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1113)
            at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4180)
            at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3769)
            at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
            at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
            at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
            at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
            at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:694)
            at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
            at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:606)
            at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
            at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
            at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
            at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
            at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
            at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
            at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
            at java.lang.reflect.Method.invoke(Unknown Source)
            at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
            at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
            at org.eclipse.equinox.launcher.Main.run(Main.java:1515)

--- a/example-generators/org.eclipse.vorto.codegen.examples.lwm2m/src/org/eclipse/vorto/codegen/examples/lwm2m/templates/FunctionBlockXmlTemplate.java
+++ b/example-generators/org.eclipse.vorto.codegen.examples.lwm2m/src/org/eclipse/vorto/codegen/examples/lwm2m/templates/FunctionBlockXmlTemplate.java
@@ -72,6 +72,10 @@ public class FunctionBlockXmlTemplate extends LWM2MConstants implements ITemplat
          final List<Object> lwm2mObjects = lwm2m.getObject();
 
          final FunctionBlock functionblock = model.getFunctionblock();
+         
+         if (functionblock == null) {
+        	 throw new RuntimeException("This model has no function block.");
+         }
 
          final Object lwm2mObject = new Object();
          lwm2mObject.setObjectType( "MODefinition" ); // waiting for Issue in
@@ -236,13 +240,13 @@ public class FunctionBlockXmlTemplate extends LWM2MConstants implements ITemplat
       item.setUnits( mappingRule.getAttributeValue(ATTRIBUTE_UNITS, "")  );
    }
 
-   private void checkResourceIdConflictAndFillSet( final Item item ) {
-      if( !resourceIdSet.isEmpty() && resourceIdSet.contains( item ) ) {
+   private void checkResourceIdConflictAndFillSet(final Item item) {
+      if (!resourceIdSet.isEmpty() && resourceIdSet.contains(item)) {
          final String errMsg = "Resource ID <" + item.getID() + "> conflict for <" + item.getName() + ">"
-            + STR_ABORT_GENERATOR;
-         throw new IllegalArgumentException( errMsg );
+		    + STR_ABORT_GENERATOR;
+		 throw new IllegalArgumentException(errMsg);
       }
-      resourceIdSet.add( item );
+	  resourceIdSet.add(item);
    }
 
    private void handleProperties( final Object lwm2mObject, final EList<Property> properties, final String operation, final InvocationContext context) {
@@ -347,7 +351,7 @@ public class FunctionBlockXmlTemplate extends LWM2MConstants implements ITemplat
    private void handleMappingRulesForProperties( final Object lwm2mObject, final Property property, final Item item, final InvocationContext context) {
       final IMapped<Property> propertyMapping = context.getMappedElement(property, STEREOTYPE_RESOURCE);
       
-      item.setID(Short.parseShort(propertyMapping.getAttributeValue(ATTRIBUTE_ID, "0")));
+      item.setID(Short.parseShort(propertyMapping.getAttributeValue(ATTRIBUTE_ID, Integer.toString(resourceIdSet.size()))));
       item.setUnits(propertyMapping.getAttributeValue(ATTRIBUTE_UNITS, ""));
       item.setOperations(propertyMapping.getAttributeValue(ATTRIBUTE_OPERATIONS, item.getOperations()));
    }

